frontend/src/metabase/public/LocaleProvider.tsx

```diff
@@ -0,0 +1,24 @@
+import { type PropsWithChildren, useEffect, useState } from "react";
```

Comment:
Some FAQs:
- Why is this outside of the sdk folder?
Because I originally wrote this to work for both embed/public and the sdk, but we scoped it down to focus on the sdk, I think it makes sense to keep it here so it's ready to be used for the core app too when neeeded.
- Why is this not a hook?
Because it could be that we realize that we need to add a loading state otherwise we render stuff with the wrong locale, instead of switching to the hook and then back, it's probably easier to keep it as a provider.
Adding the loading would make the sdk load slower, so I'd do that only if we run into the issue, we're not linked to a major release of metabase to change this so I think we can keep the code "fast" for now